src/odemis/driver/hamamatsurx.py

```diff
@@ -833,9 +845,40 @@ def updateMetadata(self, md):
 
         super(DelayGenerator, self).updateMetadata(md)
 
+    def GetPLLMode(self):
+        """
+        Get the current value from the streak unit HW.
+        :return: (bool) current streak mode value
+        """
+        pllMode_raw = self.parent.DevParamGet(self.location, "PLL Mode")  # returns a list
+        if pllMode_raw[0] == "Locked":
+            pllMode = True
+        elif pllMode_raw[0] == "Unlocked":
+            pllMode = False
+        else:
+            logging.warning("Unexpected phase-locked loop mode %s", pllMode_raw)
```

Comment:
You mentioned it has 5 options right? Some related to errors. You should expect them as well and handle them nicely. As discussed, the driver should report the errors a .state error (+ PLL mode).

To do so, I think you have 2 options:
* You have 2 functions: one which is a one-to-one mapping for the "PLL Mode" parameter, and a second "higher level" function which interprets this return value as boolean + state. 
* You have a single function which directly read this parameter and converts to boolean + state.
There should be a